web_ui/server_api.go

```diff
@@ -0,0 +1,199 @@
+	downtime, err := database.GetIncompleteDowntimes()
+	if err != nil {
+		ctx.JSON(http.StatusNotFound, server_structs.SimpleApiResp{Status: server_structs.RespFailed, Msg: "Failed to get active downtime: " + err.Error()})
+		return
+	}
+	ctx.JSON(http.StatusOK, downtime)
+}
+
+func HandleGetAllDowntime(ctx *gin.Context) {
+	downtime, err := database.GetAllDowntimes()
+	if err != nil {
+		ctx.JSON(http.StatusNotFound, server_structs.SimpleApiResp{Status: server_structs.RespFailed, Msg: "Failed to get all downtime: " + err.Error()})
+		return
+	}
+	ctx.JSON(http.StatusOK, downtime)
+}
+
+func HandleGetDowntimeByUUID(ctx *gin.Context) {
+	uuid := ctx.Param("uuid")
+	downtime, err := database.GetDowntimeByUUID(uuid)
+	if err != nil {
+		ctx.JSON(http.StatusNotFound, server_structs.SimpleApiResp{Status: server_structs.RespFailed, Msg: "Failed to get downtime by UUID: " + err.Error()})
+		return
+	}
+	ctx.JSON(http.StatusOK, downtime)
+}
+
+func HandleUpdateDowntime(ctx *gin.Context) {
+	uuid := ctx.Param("uuid")
+	var downtimeInput DowntimeInput
+	if err := ctx.ShouldBindJSON(&downtimeInput); err != nil {
+		ctx.JSON(http.StatusBadRequest, server_structs.SimpleApiResp{Status: server_structs.RespFailed, Msg: "Invalid request payload"})
+		return
+	}
+
+	if downtimeInput.Class != "" && !isValidClass(downtimeInput.Class) {
+		ctx.JSON(http.StatusBadRequest, server_structs.SimpleApiResp{Status: server_structs.RespFailed, Msg: "Invalid input downtime class"})
+		return
+	}
+	if downtimeInput.Severity != "" && !isValidSeverity(downtimeInput.Severity) {
+		ctx.JSON(http.StatusBadRequest, server_structs.SimpleApiResp{Status: server_structs.RespFailed, Msg: "Invalid input downtime severity"})
+		return
+	}
+
+	// Retrieve existing downtime record from the database
+	existingDowntime, err := database.GetDowntimeByUUID(uuid)
+	if err != nil {
+		ctx.JSON(http.StatusNotFound, server_structs.SimpleApiResp{
+			Status: server_structs.RespFailed,
+			Msg:    "Downtime record not found",
+		})
+		return
+	}
+
+	// End time should be greater than start time
+	if downtimeInput.EndTime != 0 && downtimeInput.EndTime <= downtimeInput.StartTime {
+		ctx.JSON(http.StatusBadRequest, server_structs.SimpleApiResp{
+			Status: server_structs.RespFailed,
+			Msg:    "End time should be greater than start time",
+		})
+		return
+	}
+
+	// Only update fields provided in the request
+	if downtimeInput.CreatedBy != "" {
+		existingDowntime.CreatedBy = downtimeInput.CreatedBy
+	}
+	if downtimeInput.Class != "" {
+		existingDowntime.Class = downtimeInput.Class
+	}
+	if downtimeInput.Description != "" {
+		existingDowntime.Description = downtimeInput.Description
+	}
+	if downtimeInput.Severity != "" {
+		existingDowntime.Severity = downtimeInput.Severity
+	}
+	if downtimeInput.StartTime != 0 {
+		existingDowntime.StartTime = downtimeInput.StartTime
+	}
+	if downtimeInput.EndTime != 0 {
+		existingDowntime.EndTime = downtimeInput.EndTime
```

Comment:
```suggestion
		existingDowntime.EndTime = downtimeInput.EndTime
```

Won't this break if I want to update the EndTime value to 0? 0 being a valid value indicating this does not end.